lib/ddtrace/sampling/rule_sampler.rb

```diff
@@ -0,0 +1,77 @@
+require 'forwardable'
+
+require 'ddtrace/ext/priority'
+
+require 'ddtrace/ext/sampling'
+require 'ddtrace/sampler'
+require 'ddtrace/sampling/token_bucket'
+
+module Datadog
+  module Sampling
+    # TODO: Write class documentation
+    # RuleSampler
+    class RuleSampler
+      extend Forwardable
+
+      attr_reader :rules, :rate_limiter, :fallback_sampler
+
+      def initialize(rules = [],
+                     rate_limiter = Datadog::Sampling::UnlimitedLimiter.new,
+                     fallback_sampler = Datadog::AllSampler.new)
+        @rules = rules
+        @rate_limiter = rate_limiter
+        @fallback_sampler = fallback_sampler
+      end
+
+      # /RuleSampler's components (it's rate limiter, for example) are
+      # not be guaranteed to be size-effect free.
+      # It is not possible to guarantee that a call to {#sample?} will
+      # return the same result as a successive call to {#sample!} with the same span.
+      #
+      # Use {#sample!} instead
+      def sample?(_span)
+        raise 'RuleSampler cannot be evaluated without side-effects'
+      end
+
+      def sample!(span)
+        sampled = sample_span(span) { |s| @fallback_sampler.sample!(s) }
+
+        sampled.tap do
+          span.sampled = sampled
+        end
+      end
+
+      def_delegators :@fallback_sampler, :update
```

Comment:
we can `no-op` `:update`, we don't need to consider agent returned sample rates at all here.

Reply:
This is effectively a no-op, as most sampler implementations we delegate this call to are no-ops.

I think we should keep this delegate as we currently support `RateByServiceSampler` in our app. We could remove `update` it if we drop support for it in the future.